src/tools/clippy/clippy_lints/src/eta_reduction.rs

```diff
@@ -200,7 +205,13 @@ fn check_inputs(cx: &LateContext<'_>, params: &[Param<'_>], call_args: &[Expr<'_
             },
             _ => false,
         }
-    })
+    };
+    if let Some(receiver) = receiver {
+        std::iter::zip(params, std::iter::once(receiver).chain(call_args.iter()))
+            .all(|(param, arg)| check_inputs(param, arg))
+    } else {
+        std::iter::zip(params, call_args).all(|(param, arg)| check_inputs(param, arg))
+    }
```

Comment:
Can we merge both branches with a `receiver.into_iter().chain(call_args.iter())`?